### PR TITLE
Use nullptr in core classes

### PR DESCRIPTION
**Pull request by linev**
Avoid warnings when -Wzero-as-null-pointer-constant is used

### CODE REVIEW

**Comment by Axel-Naumann** [comment]
Could you move the changes in `clingutils` and `dictgen` into a separate PR that we look at once the llvm upgrade is in? I'm afraid of the conflicts we get from this PR.

**Comment by linev** [comment]
> Could you move the changes in clingutils and dictgen into a separate PR

I move them to #11351 - to have them in single place

File: core/meta/inc/TDictionary.h [hunk 1]
```diff
@@ -167,15 +167,15 @@ namespace ROOT {
...
+   TDictAttributeMap *fAttributeMap{nullptr};    //pointer to a class attribute map
+   ULong64_t fUpdatingTransactionCount{0}; //!the Cling ID of the transaction that last updated the object
 
 protected:
    Bool_t              UpdateInterpreterStateMarker();
 
 public:
-   TDictionary(): fAttributeMap(0), fUpdatingTransactionCount(0) { }
-   TDictionary(const char* name): TNamed(name, ""), fAttributeMap(0), fUpdatingTransactionCount(0) { }
+   TDictionary(): fAttributeMap(nullptr), fUpdatingTransactionCount(0) { }
```
Comment (pcanal) [inline core/meta/inc/TDictionary.h:177]:
```suggestion
   TDictionary() = default;
```

**Comment by pcanal** [comment]
Thanks Sergey.  Can you also open the required roottest merge request?

File: core/meta/inc/TDictionary.h (hunk 1 above)
Comment (linev) [reply core/meta/inc/TDictionary.h:177]: Philippe, it is not allowed to use `default` constructor for `TObject` - derived classes.
Here more details: https://sft.its.cern.ch/jira/browse/ROOT-10300

Comment (pcanal) [reply core/meta/inc/TDictionary.h:177]: right, I forgot :(

**Review by pcanal — APPROVED** [review]

**Pull request merged**